tests/jiva/clone/provision_test.go

```diff
@@ -156,11 +209,113 @@ var _ = Describe("[jiva] TEST JIVA CLONE CREATION", func() {
 		})
 	})
 
+	When("creating application pod with above pvc as volume", func() {
+		It("should create a running pod", func() {
+			clonePodObj, err = pod.NewBuilder().
+				WithName(cloneAppName).
+				WithNamespace(namespaceObj.Name).
+				WithContainerBuilder(
+					container.NewBuilder().
+						WithName("busybox").
+						WithImage("busybox").
+						WithCommand(
+							[]string{
+								"sh",
+								"-c",
+								"tail -f /dev/null",
+							},
+						).
+						WithVolumeMounts(
+							[]corev1.VolumeMount{
+								corev1.VolumeMount{
+									Name:      "demo-vol1",
+									MountPath: "/mnt/store1",
+								},
+							},
+						),
+				).
+				WithVolumeBuilder(
+					volume.NewBuilder().
+						WithName("demo-vol1").
+						WithPVCSource(cloneName),
+				).
+				Build()
+			Expect(err).ShouldNot(HaveOccurred(), "while building pod {%s}", cloneAppName)
+
+			By("creating pod with above pvc as volume")
+			clonePodObj, err = ops.PodClient.WithNamespace(namespaceObj.Name).Create(clonePodObj)
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while creating pod {%s} in namespace {%s}",
+				cloneAppName,
+				namespaceObj.Name,
+			)
+
+			By("verifying pod is running")
+			status := ops.IsPodRunningEventually(namespaceObj.Name, cloneAppName)
+			Expect(status).To(Equal(true), "while checking status of pod {%s}", cloneAppName)
+
+		})
+	})
+
+	When("verifying data consistency in pvc and clone pvc", func() {
+		It("should have consistent data between the two pvcs", func() {
+			By("fetching data from original pvc")
+			podOutput, err := ops.PodClient.WithNamespace(namespaceObj.Name).
+				Exec(
+					podObj.Name,
+					&corev1.PodExecOptions{
+						Command: []string{
+							"sh",
+							"-c",
+							"md5sum mnt/store1/date.txt",
+						},
+						Container: "busybox",
+						Stdin:     false,
+						Stdout:    true,
+						Stderr:    true,
+					},
+				)
+			Expect(err).ShouldNot(HaveOccurred(), "while exec in application pod")
+
+			By("fetching data from clone pvc")
+			clonePodOutput, err := ops.PodClient.WithNamespace(namespaceObj.Name).
+				Exec(
+					clonePodObj.Name,
+					&corev1.PodExecOptions{
+						Command: []string{
+							"sh",
+							"-c",
+							"md5sum mnt/store1/date.txt",
+						},
+						Container: "busybox",
+						Stdin:     false,
+						Stdout:    true,
+						Stderr:    true,
+					},
+				)
+			Expect(err).ShouldNot(HaveOccurred(), "while exec in clone application pod")
+
+			By("veryfing data consistency")
+			Expect(podOutput).To(Equal(clonePodOutput), "while checking data consistency")
```

Comment:
Can we also do a negative test case where clone does not have the data available in original pod. This data should be persisted after taking a clone.

Reply:
So in negative test the pvc can have incorrect storage class or the application have invalid volume or invalid command. What should we test for ? There are many ways to amke data non persistent. Which one to pick ?

Reply:
ok.. so this is not exactly negative case. Its a case where a file was created on orig volume after taking the snapshot.
now when a clone gets created out of above snapshot this clone will not have this new file.